plugin-build/src/main/kotlin/io/sentry/android/gradle/telemetry/SentryTelemetryService.kt

```diff
@@ -0,0 +1,474 @@
+import org.gradle.process.ExecOperations
+import org.gradle.util.GradleVersion
+
+abstract class SentryTelemetryService :
+    BuildService<None>,
+    BuildOperationListener,
+    AutoCloseable {
+
+    private var hub: IHub = NoOpHub.getInstance()
+    private var transaction: ITransaction? = null
+    private var didAddChildSpans: Boolean = false
+    private var started: Boolean = false
+
+    fun start(startParameters: SentryTelemetryServiceParams) {
+        if (started) {
+            return
+        }
+
+        started = true
+        try {
+            if (startParameters.saas == false) {
+                SentryPlugin.logger.info(
+                    "Sentry is running against a self hosted instance. Telemetry has been disabled."
+                )
+                hub = NoOpHub.getInstance()
+            } else if (!startParameters.sendTelemetry) {
+                SentryPlugin.logger.info("Sentry telemetry has been disabled.")
+                hub = NoOpHub.getInstance()
+            } else {
+                if (!Sentry.isEnabled()) {
+                    SentryPlugin.logger.info(
+                        "Sentry telemetry is enabled. To disable set `telemetry=false` " +
+                            "in the sentry config block."
+                    )
+                    Sentry.init { options ->
+                        options.dsn = startParameters.dsn
+                        options.isDebug = true
+                        options.isEnablePrettySerializationOutput = false
+                        options.tracesSampleRate = 1.0
+                        options.release = BuildConfig.Version
+                        options.isSendModules = false
+                        options.environment = startParameters.buildType
+                        options.setTag("SDK_VERSION", BuildConfig.SdkVersion)
+                        options.setTag("AGP_VERSION", AgpVersions.CURRENT.toString())
+                        options.setTag("BUILD_SYSTEM", "gradle")
+                        options.setTag("BUILD_TYPE", startParameters.buildType)
+                        options.setTag("GRADLE_VERSION", GradleVersion.current().version)
+                        startParameters.cliVersion?.let { options.setTag("SENTRY_CLI_VERSION", it) }
+
+                        startParameters.extraTags.forEach { (key, value) ->
+                            options.setTag(
+                                key,
+                                value
+                            )
+                        }
+                    }
+                }
+                hub = Sentry.getCurrentHub()
+                startRun("gradle build ${startParameters.buildType}")
+
+                hub.configureScope { scope ->
+                    scope.user = User().also { user ->
+                        startParameters.defaultSentryOrganization.let { user.id = it }
+                        startParameters.sentryOrganization.let { user.id = it }
```

Comment:
```suggestion
                        startParameters.defaultSentryOrganization?.let { user.id = it }
                        startParameters.sentryOrganization?.let { user.id = it }
```
shouldn't it be like this rather?